microfs.py

```diff
@@ -56,10 +56,16 @@ def raw_on(serial):
     """
     Puts the device into raw mode.
     """
-    serial.write(b'\x03')  # Send CTRL-C to break out of loop.
-    serial.read_until(b'\n>')  # Flush buffer until prompt.
-    serial.write(b'\x01')  # Go into raw mode.
-    serial.read_until(b'\r\n>OK')  # Flush buffer until raw mode prompt.
+    attempts_left = 3
+    while attempts_left:
+        serial.write(b'\x03')    # Send CTRL-C to break out of loop.
+        if serial.read_until(b'\n>>>').endswith(b'\n>>>'):
+            break
+        attempts_left -= 1
+    else:
+        raise IOError('Could not enter REPL mode.')
+    serial.write(b'\x01')        # Go into raw mode.
+    serial.read_until(b'\r\n>')  # Flush buffer until raw mode prompt.
```

Comment:
The original line here was reading until `b'\r\n>OK'`, but in my tests MicroPython wasn't send the last `OK`, this is what I get:
```
\r\nraw REPL; CTRL-B to exit\r\n>
```
In this old code `read_until` worked because it also returns when no more characters are received.

Reply:
Ahh, this is good to know.

Reply:
That's right, uPy won't send an `OK` unless the raw REPL receives some data to process.  When you enter raw REPL with `b'\x01'` you shouldn't be waiting for an `OK`.

Reply:
OK... I *DO* get an `OK` on Linux.

Since this *was* working I'll revert to the old way. :-/

Reply:
The `OK` should only be sent after a `b'\x04` (Ctrl+D), that's a bit weird as well.